go/cmd/dolt/commands/schcmds/import.go

```diff
@@ -16,26 +16,313 @@ package schcmds
+	ap.SupportsString(mappingParam, "", "mapping-file", "A file that can map a column name in <file> to a new value.")
+	ap.SupportsString(floatThresholdParam, "", "float", "Minimum value at which the fractional component of a value must exceed in order to be considered a float.")
+	ap.SupportsString(delimParam, "", "delimiter", "Specify a delimiter for a csv style file with a non-comma delimiter.")
 
 	help, usage := cli.HelpAndUsagePrinters(commandStr, schImportShortDesc, schImportLongDesc, schImportSynopsis, ap)
-	cli.ParseArgs(ap, args, help)
+	apr := cli.ParseArgs(ap, args, help)
+
+	if apr.NArg() != 2 {
+		usage()
+		return 0
+	}
+
+	verr := importSchema(ctx, dEnv, apr)
+
+	return commands.HandleVErrAndExitCode(verr, usage)
+}
+
+func importSchema(ctx context.Context, dEnv *env.DoltEnv, apr *argparser.ArgParseResults) errhand.VerboseError {
+	root, verr := commands.GetWorkingWithVErr(dEnv)
+
+	if verr != nil {
+		return verr
+	}
+
+	tblName := apr.Arg(0)
+	fileName := apr.Arg(1)
+
+	fileExists, _ := dEnv.FS.Exists(fileName)
+
+	if !fileExists {
+		return errhand.BuildDError("error: file '%s' not found.", fileName).Build()
+	}
+
+	op := createOp
+	if !apr.ContainsAny(createFlag, updateFlag, replaceFlag) {
+		return errhand.BuildDError("error: missing required parameter.").AddDetails("Must provide exactly one of the operation flags '--create', or '--replace'").SetPrintUsage().Build()
+	} else if apr.Contains(updateFlag) {
+		if apr.ContainsAny(createFlag, replaceFlag) {
+			return errhand.BuildDError("error: multiple operations supplied").AddDetails("Only one of the flags '--create', '--update', or '--replace' may be provided").SetPrintUsage().Build()
+		}
+		op = updateOp
+	} else if apr.Contains(replaceFlag) {
+		if apr.Contains(createFlag) {
+			return errhand.BuildDError("error: multiple operations supplied").AddDetails("Only one of the flags '--create', '--update', or '--replace' may be provided").SetPrintUsage().Build()
+		}
+		op = replaceOp
+	} else {
+		if apr.Contains(keepTypesParam) {
+			return errhand.BuildDError("error: parameter keep-types not supported for create operations").AddDetails("keep-types parameter is used to keep the existing column types as is without modification.").Build()
+		}
+	}
+
+	tbl, tblExists, err := root.GetTable(ctx, tblName)
+
+	if err != nil {
+		return errhand.BuildDError("error: failed to read from database.").AddCause(err).Build()
+	} else if tblExists && op == createOp {
+		return errhand.BuildDError("error: failed to create table.").AddDetails("A table named '%s' already exists.", tblName).AddDetails("Use --replace or --update instead of --create.").Build()
+	}
+
+	var existingSch schema.Schema = schema.EmptySchema
+	if tblExists {
+		existingSch, err = tbl.GetSchema(ctx)
+
+		if err != nil {
+			return errhand.BuildDError("error: failed to read schema from '%s'", tblName).AddCause(err).Build()
+		}
+	}
+
+	val, pksOK := apr.GetValue(pksParam)
```

Comment:
Same for this and other uses of apr below, don't even pass apr into this function